pvlib/pvsystem.py

```diff
@@ -1965,25 +1969,22 @@ def pvsyst_celltemp(poa_global, wind_speed, temp_air, eta_m=0.1,
     photovoltaic modules." Progress in Photovoltaics 16(4): 307-315.
     """
 
-    temp_models = TEMP_MODELS['pvsyst']
+    pvsyst_defaults = {"freestanding": (29.0, 0), "insulated": (15.0, 0)}
```

Comment:
Not OK with this change.  We want to group the sets of coefficients into a common dict outside of the function so that it is easier to write an `infer` method for `ModelChain`

Reply:
@adriesse what was your reasoning for inlining the defaults here?

@cwhanse I don't think this makes any difference for `ModelChain`, at least at this point in time. I think the idea was to allow people to inspect and modify the dictionary programmatically instead of manually - which could be good or bad.

Reply:
I think the version of `pvsystem.py` I started with had this line, and I merely changed the name.  But I see where you're coming from with the grouping.  How easy is it for a use to find out what the available options are from your global list?

Reply:
I see `infer_xxx` more as a work-around generally.  Isn't there some kind of dictum about explicit being better than implicit?  :)

Reply:
The idea of `infer_` at the `ModelChain` level is to let a user specify a racking model (one of the available keys in `TEMP_MODELS`) for a PVSystem instance and pvlib will figure out which cell temperature model to use.

Reply:
IMO there should be a distinction between porcelain and plumbing. Stop me if I'm preaching to the choir.

### plumbing
Low level functions should be explicit, arguments should be all spelled out, more primitive and native as you go lower level, features should simple

### porcelain
High level classes and functions should make things easy for the user. This may mean that they **_infer_** the users desire, they have preset defaults, they do lots of fancy magic without the user having to understand all the details

Reply:
I certainly understand and appreciate the intent. Is there is a difference from the DC_MODEL_PARAMS situation?  It seems that there you're detecting parameter *names*, whereas with the temperature models you're planning to detect parameter *values*.

Reply:
You are correct, with DC_MODEL_PARAMS we're inferring a DC model from the field names, with TEMP_MODELS we're inferring a cell temperature model from the racking name string.  TEMP_MODELS is a bit clunky, and could be improved with a parallel to DC_MODEL_PARAMS, and a new function that translates from the racking strings to parameter sets.